lib/Core/Searcher.h

```diff
@@ -166,11 +167,27 @@ namespace klee {
     }
   };
 
+  /* RandomPathSearcher performs a random walk of the PTree to
+     select a state. PTree is a global data structure, however
+     a searcher can sometimes only select from a subset of all states
+     (depending on the update calls).
+
+     To support this RP has a subgraph view of PTree, in that it only
+     walks the PTreeNodes that it "owns". Ownership is stored in the
+     getInt method of PTreeNodePtr class (which hides it in the pointer itself).
+
+     The ownership bits are maintained in the update method.
+  */
   class RandomPathSearcher : public Searcher {
-    Executor &executor;
+    PTree &processTree;
+    // Number of states handled by this searcher
+    int size = 0;
```

Comment:
`int` as state counter supports less than a month KLEE run-time on COTS hardware for many benchmarks...

Reply:
I strongly doubt you can find hardware, where KLEE will have 2 billion states, but sure.

Reply:
That's just ~600GB of `ExecutionState`s... at least make it `uint32_t`.

@MartinNowack I have an idea for new server hardware. :sunglasses:
https://www.dell.com/en-uk/work/shop/povw/poweredge-r940
```
Maximum RAM
RDIMM 3TB
LRDIMM 6TB 
NVDIMM 384GB
DCPMM 12.28TB (15.36 with LRDIMM)
```

Reply:
wait, that's only 300B per state? with 3GB limit, it's quite good to see 100 000 states. That's 30KB per state. For 2 billion states, that's 60TB, so still too small, but quite close actually :O

Reply:
A single `ExecutionState` object is a little bit above 300 bytes I think. This does not include all the other bookkeeping data (caches, constraints, stacks...).

Reply:
I think my 30KB estimate is more realistic.

Reply:
For Coreutils it's between 13kB and 1.2MB in my last experiment - assuming that all of the memory is used for state handling.